src/ixdat/techniques/cv.py

```diff
@@ -107,3 +122,182 @@ def redefine_cycle(self, start_potential=None, redox=None):
+
+        There are three types: "anodic" (positive scan rate), "cathodic" (negative scan
+        rate), and "hold" (zero scan rate)
+        """
+        t = self.t
+        ec_sweep_types = {
+            "positive": "anodic",
+            "negative": "cathodic",
+            "steady": "hold",
+        }
+        indexed_sweeps = find_signed_sections(self.scan_rate.data)
+        timed_sweeps = []
+        for (i_start, i_finish), general_sweep_type in indexed_sweeps:
+            timed_sweeps.append(
+                ((t[i_start], t[i_finish]), ec_sweep_types[general_sweep_type])
+            )
+        return timed_sweeps
+
+    def diff_with(self, other, v_list=None, cls=None):
+        """Return a CyclicVotammagramDiff of this CyclicVotammagram with another one
+
+        Each anodic and cathodic sweep in other is lined up with a corresponding sweep
+        in self. Each variable given in v_list (defaults to just "current") is
+        interpolated onto self's potential and subtracted from self.
+        """
+
+        vseries = self.potential
+        tseries = vseries.tseries
+        series_list = [tseries, self.raw_potential, self.cycle]
+
+        v_list = v_list or ["current", "raw_current"]
+        if "potential" in v_list:
+            raise BuildError(
+                f"v_list={v_list} is invalid. 'potential' is used to interpolate."
+            )
+
+        my_sweep_specs = [
+            spec for spec in self.sweep_specs if spec[1] in ["anodic", "cathodic"]
+        ]
+        others_sweep_specs = [
+            spec for spec in other.sweep_specs if spec[1] in ["anodic", "cathodic"]
+        ]
+        if not len(my_sweep_specs) == len(others_sweep_specs):
+            raise BuildError(
+                "Can only make diff of CyclicVoltammagrams with same number of sweeps."
+                f"{self} has {my_sweep_specs} and {other} has {others_sweep_specs}."
+            )
+
+        diff_values = {name: np.array([]) for name in v_list}
+        t_diff = np.array([])
+
+        for my_spec, other_spec in zip(my_sweep_specs, others_sweep_specs):
+            sweep_type = my_spec[1]
+            if not other_spec[1] == sweep_type:
+                raise BuildError(
+                    "Corresponding sweeps must be of same type when making diff."
+                    f"Can't align {self}'s {my_spec} with {other}'s {other_spec}."
+                )
+            my_tspan = my_spec[0]
+            other_tspan = other_spec[0]
+            my_t, my_potential = self.grab(
+                "potential", my_tspan, include_endpoints=False
+            )
+            t_diff = np.append(t_diff, my_t)
+            other_t, other_potential = other.grab(
+                "potential", other_tspan, include_endpoints=False
+            )
+            if sweep_type == "anodic":
+                other_t_interp = np.interp(
+                    np.sort(my_potential), np.sort(other_potential), other_t
```

Comment:
I get a ValueError: array of sample points is empty if I try to subtract two identical CVs.

Reply:
Oh, wierd. That will take some debugging. Subtracting two identical CV's should, in my opinion, give a CyclicVoltammagramDiff with the curent equal to zero.